cyvcf2/tests/test_reader.py

```diff
@@ -669,6 +657,31 @@ def test_set_format_int3():
 
     assert np.allclose(v.format("P3"), exp)
 
+def test_set_format_str_bytes_second_longer():
+    vcf = VCF('{}/test-format-string.vcf'.format(HERE))
+    assert vcf.add_format_to_header(dict(ID="STR", Number=1, Type="String", Description="String example")) == 0
+    v = next(vcf)
+
+    v.set_format("STR", np.array([b'foo', b'barbaz']))
+    assert np.all(v.format('STR') == np.array(['foo', 'barbaz']))
+
+def test_set_format_str_bytes_first_longer():
+    vcf = VCF('{}/test-format-string.vcf'.format(HERE))
+    assert vcf.add_format_to_header(dict(ID="STR", Number=1, Type="String", Description="String example")) == 0
+    v = next(vcf)
+
+    v.set_format("STR", np.array([b'foobar', b'baz']))
+    assert np.all(v.format('STR') == np.array(['foobar', 'baz']))
+
+def test_set_format_str_bytes_number3():
```

Comment:
so this test is failing?

Reply:
I'm not exactly sure what you mean by that, so to be completely pedantic: the test `test_set_format_str_bytes_number3` reliably generates an exception in an expected manner because the code now raises an error when you try to set an array of strings with len(shape) > 1. Is that what you meant?

Reply:
bah. I missed that it was actually testing the exception. very thorough. thanks.